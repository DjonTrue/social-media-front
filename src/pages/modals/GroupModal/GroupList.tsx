import React from "react";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import CloseIcon from "@material-ui/icons/Close";
import { useDeleteGroupMutation, useGetGroupsQuery } from "./groupModalApi";

const GroupList = () => {
  const { data: groups } = useGetGroupsQuery("20");
  const [deleteGroup] = useDeleteGroupMutation();

  const handleDeleteGroup = (id: number) => {
    deleteGroup(id);
  };

  return (
    <>
      <h3>{groups && `${groups.length} groups are avialeble for you`}</h3>
      {groups &&
        groups.map(({ groupName, groupDesc, authorName, id }: any) => (
          <ListItem key={id} component="div" disablePadding>
            <ListItemButton
              style={{
                width: "100%",
                display: "flex",
                flexDirection: "column",
                lineHeight: ".1rem",
              }}
            >
              <h3>Name {groupName || "Not initialized"}</h3>
              <h4>Author: {authorName || "Unknown author"}</h4>
              <p>Description {groupDesc}</p>
            </ListItemButton>
            <ListItemButton>
              <CloseIcon style={{ cursor: "pointer" }} onClick={() => handleDeleteGroup(id)} />
            </ListItemButton>
          </ListItem>
        ))}
    </>
  );
};

export default GroupList;
