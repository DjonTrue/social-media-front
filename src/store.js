import { configureStore } from "@reduxjs/toolkit";
import { modalReducer } from "./features/groupModal/groupModal";
import registrationReducer from "./features/Auth/registrationSlice";
import { postReducer } from "./features/fetchingPosts/fetchingPostsSlice";
import { groupModalApi } from "./pages/modals/GroupModal/groupModalApi";

// This approach is useful for generate specific store for tests
export const createReduxStore = (initialState = {}) => {
  return configureStore({
    reducer: {
      registration: registrationReducer,
      posts: postReducer,
      modal: modalReducer,
      [groupModalApi.reducerPath]: groupModalApi.reducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(groupModalApi.middleware),
    preloadedState: initialState,
  });
};
