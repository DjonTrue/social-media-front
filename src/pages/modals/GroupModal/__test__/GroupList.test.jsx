import React from "react";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import GroupList from "../GroupList";
import { createReduxStore } from "../../../../store";

describe("Test GroupList component renders", () => {
  it("PrimaryButton component renders fine", () => {
    render(
      <Provider store={createReduxStore()}>
        <GroupList />
      </Provider>
    );
    const btn = screen.queryByRole("button");

    expect(btn).not.toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });
});
