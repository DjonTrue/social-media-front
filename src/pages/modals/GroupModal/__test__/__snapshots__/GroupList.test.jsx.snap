// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GroupList component renders PrimaryButton component renders fine 1`] = `null`;
